inc/classes/class-srm-post-type.php

```diff
@@ -252,6 +252,50 @@ public function action_redirect_chain_alert() {
 				</div>
 				<?php
 			}
+
+			$duplicate_redirect_rule = filter_input( INPUT_GET, 'duplicate-redirect-rule', FILTER_VALIDATE_INT );
+
+			if ( ! empty( $duplicate_redirect_rule ) ) {
+
+				?>
+				<div class="notice notice-error is-dismissible">
+					<p>
+						<?php esc_html_e( 'Safe Redirect Manager Error: Duplicate redirection rule found. You can modify below rule instead of adding new one.', 'safe-redirect-manager' ); ?>
+					</p>
+				</div>
+				<?php
+
+				$should_delete = true;
+				$_wpnonce      = filter_input( INPUT_GET, '_wpnonce', FILTER_SANITIZE_STRING );
+				$error         = '';
+
+				if ( ! wp_verify_nonce( $_wpnonce, 'delete-duplicate-redirect-rule' ) ) {
+
+					$should_delete = false;
+					$error         = __( 'Security check failed when deleting duplicate redirect rule, please contact your administrator!', 'safe-redirect-manager' );
+
+				} elseif ( ! current_user_can( 'delete_post', $duplicate_redirect_rule ) ) {
+
+					$should_delete = false;
+					$error         = __( 'You don\'t have enough permission to delete duplicate redirect rule, please contact your administrator!', 'safe-redirect-manager' );
+				}
+
+				if ( $should_delete ) {
+
+					// Delete duplicate post.
+					wp_delete_post( $duplicate_redirect_rule, true );
```

Comment:
@sanketio Deleting a post where the ID is passed in a callback seems like a security risk vector I'd love to avoid introducing if possible. 

I'm wondering if this approach could be shifted to acting at a different point, maybe before the rule is even saved? The idea would be:

* On the "Add Redirect" screen, the "Publish" and "Save Draft" buttons would start disabled. 
* As the user (paused/debounced/cached) types into the Redirect From field, send an REST Ajax request to and endpoint that checks for existing matching rules.
* If there is no match, the rule can be published/saved - the buttons become enabled.
* If there is a match, the buttons remain disabled & a warning message appears with a link to "Edit existing rule" (returned by the endpoint if needed).

I realize this adds some code overhead, however if feels like this would result in both a better user experience and a less risky endpoint (no more arbitrary post deleting!) - what do you think?

One more thing: do we want to try to account for wildcards at all? what happens if I have an existing rule for `/foo/*` and I try to add a rule for `foo/bar/` - does that work correctly?